### PR TITLE
feat(component): button, input, textarea, select enhanced dt7 styles

### PR DESCRIPTION
**Pull request by francisrupert**
## Description

Components Buttons, Input, Textarea, Select: CSS-only updates to enhance DT7 alignment.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

image

### CODE REVIEW

File: docs/.vuepress/theme/components/Navbar.vue [hunk 1]
```diff
@@ -17,7 +17,7 @@
     <div class="d-d-flex d-flow4">
       <a
         alt="GitHub repo"
-        class="d-btn d-btn--muted d-tooltip--hover"
+        class="d-btn d-btn--muted d-btn--circle d-tooltip--hover"
```
Comment (braddialpad) [inline docs/.vuepress/theme/components/Navbar.vue:20]: Was it intentional to make the padding on the header this tight? Fine if it was, just checking it wasn't an unintended side-effect.

before:
Screenshot 2022-12-21 at 9 47 46 AM

after:
Screenshot 2022-12-21 at 9 47 36 AM

File: lib/build/less/components/button.less [hunk 2]
```diff
@@ -17,34 +17,89 @@
 //  ============================================================================
 //  $   BASE STYLE
 //  ----------------------------------------------------------------------------
-.d-btn {
+
+.asdfasdf {
```
Comment (braddialpad) [inline lib/build/less/components/button.less]: `.asdfasdf`? was this for testing or something?

File: lib/build/less/components/button.less [hunk 3]
```diff
@@ -187,36 +254,17 @@
...
-.d-btn__icon--top ~ .d-btn__label,
-.d-btn__label ~ .d-btn__icon--bottom {
-    margin-top: var(--space-300);
-}
-
-.d-btn__icon--bottom ~ .d-btn__label,
-.d-btn__label ~ .d-btn__icon--top {
-    margin-bottom: var(--space-300);
-}
-
```
Comment (braddialpad) [inline lib/build/less/components/button.less:209]: I think I tried to remove this before and it broke something legacy code related.. We'll do some testing to confirm.

File: docs/components/button.md [hunk 4]
```diff
@@ -6,6 +6,7 @@ thumb: true
 storybook_url: https://vue.dialpad.design/?path=/story/components-button--default
```
Comment (braddialpad) [inline docs/components/button.md:6]: Looks like the correct border radius is not applied to the xs size

Screenshot 2022-12-21 at 10 02 05 AM

File: docs/components/input.md [hunk 5]
```diff
@@ -6,7 +6,8 @@ thumb: true
 storybook_url: https://vue.dialpad.design/?path=/story/components-input--default
```
Comment (braddialpad) [inline docs/components/input.md:6]: Looks like this was a pre-existing issue, but might as well fix it now while we're making all these changes. There is no focus ring when clicking on the warning validation state input. For success and error there is.

### Success input focused
Screenshot 2022-12-21 at 10 09 30 AM

### Warning input focused.
Screenshot 2022-12-21 at 10 09 38 AM

File: docs/.vuepress/views/IconCatalog.vue [hunk 6]
```diff
@@ -156,5 +159,8 @@ const selectIcon = (index) => {
 </script>
 
 <style scoped>
-
+  /* more or less a hack, 🤷‍♂️ */
+  #search-input-button-close {
+    margin-right: calc(var(--size-300) * -1)
+  }
```
Comment (braddialpad) [inline docs/.vuepress/views/IconCatalog.vue:165]: Hmm was the default rightIcon position not sufficient?

**Review by braddialpad — COMMENTED** [review]
Lookin good!

File: docs/.vuepress/theme/components/Navbar.vue (hunk 1 above)
Comment (francisrupert) [reply docs/.vuepress/theme/components/Navbar.vue:20]: Not intentional. Fixed.

File: lib/build/less/components/button.less (hunk 2 above)
Comment (francisrupert) [reply lib/build/less/components/button.less]: yeah, removed; a few others too.

File: docs/components/button.md (hunk 4 above)
Comment (francisrupert) [reply docs/components/button.md:6]: Actually it's intentional, radius and spacing proportionally increases. Although... now that I look at it... I might prefer it be same radius as the next bigger size.

File: docs/.vuepress/views/IconCatalog.vue (hunk 6 above)
Comment (francisrupert) [reply docs/.vuepress/views/IconCatalog.vue:165]: It was for just an icon, but wrapping the icon with a circle button indented it too much. I figure having its CSS scoped here is fine for now.

File: docs/components/button.md (hunk 4 above)
Comment (francisrupert) [reply docs/components/button.md:6]: Explored it. Leaving alone for now. We'll be rebuilding all of button/input/select in Q1'23 anyhow.

File: docs/components/input.md (hunk 5 above)
Comment (francisrupert) [reply docs/components/input.md:6]: fixed. "yellow" wasn't replaced with "gold"

**Review by braddialpad — APPROVED** [review]
Thanks!

**Pull request merged**